plugin/src/test/java/com/redhat/jenkins/plugins/ci/integration/AmqMessagingPluginIntegrationTest.java

```diff
@@ -489,4 +497,33 @@ public void testSimpleCIEventWithMessagePropertiesAsVariable() throws Exception
         j.assertLogContains("echo TEST_PROP1 = GOT 1", lastBuild);
         j.assertLogContains("echo TEST_PROP2 = GOT 2", lastBuild);
     }
+
+    @Test
+    public void testTTL() throws Exception {
+        FreeStyleProject jobA = j.createFreeStyleProject();
+        jobA.getBuildersList().add(new Shell("echo $MESSAGE_HEADERS"));
+        attachTrigger(new CIBuildTrigger(false, Collections.singletonList(getSubscriberProviderData(
+                null, null, "CI_TYPE = 'code-quality-checks-done' and CI_STATUS = 'failed'"
+        ))), jobA);
+
+        FreeStyleProject jobB = j.createFreeStyleProject();
+        jobB.getPublishersList().add(new CIMessageNotifier(getPublisherProviderData(
+                null, MessageUtils.MESSAGE_TYPE.CodeQualityChecksDone, "CI_STATUS = failed", null
+        )));
+        
+        j.buildAndAssertSuccess(jobB);
+        waitUntilScheduledBuildCompletes();
+        j.assertBuildStatusSuccess(jobA.getLastBuild());
+        j.assertLogContains("\"JMSExpiration\":0", jobA.getLastBuild());
+
+        jobB.getPublishersList().add(new CIMessageNotifier(getPublisherProviderData(
+                null, MessageUtils.MESSAGE_TYPE.CodeQualityChecksDone, "CI_STATUS = failed", null, 1000
+        )));
+
+        j.buildAndAssertSuccess(jobB);
+        waitUntilScheduledBuildCompletes();
+        j.assertBuildStatusSuccess(jobA.getLastBuild());
+        j.assertLogNotContains("JMSExpiration: 0", jobA.getLastBuild());
+        j.assertLogContains("\"JMSExpiration\":", jobA.getLastBuild());
```

Comment:
do you not want to check for `1000` here?

Reply:
No, because the JMSExpiration actually has a timestamp.
```
Message Headers:
  JMSDestination: topic://VirtualTopic.qe.ci.>
  JMSDeliveryMode: persistent
  JMSExpiration: 14:20:37
  JMSPriority: 4
  JMSMessageID: ID:jenkins-2-9ldm7-34257-1671630733009-11:1:1:1:1
  JMSTimestamp: 14:20:33
  JMSCorrelationID: null
  JMSReplyTo: null
  JMSRedelivered: false
  JMSType: application/json
Message Properties:
  CI_NAME: send-message-pipeline
  CI_TYPE: code-quality-checks-done
  foo: bar
Message Body:
some content
```

Reply:
Correct, 0 never expires and is the default.